jdat_notebooks/preimaging/preimaging_02_calwebb.ipynb

```diff
@@ -0,0 +1,279 @@
+    "We select the *_uncal.fits files and execute calwebb_detector1 to generate the *_rate.fits files."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "for input_yaml_path in inputyaml:\n",
+    "    print(input_yaml_path) \n",
+    "    \n",
+    "    base = os.path.basename(input_yaml_path)\n",
+    "    name = os.path.splitext(base)[0]\n",
+    "\n",
+    "    exi = os.path.exists(name + '_rate.fits')\n",
+    "    if exi:\n",
+    "        continue\n",
+    "        \n",
+    "    namelist.append(name + '_uncal.fits')\n",
+    "    outlist.append(name + '.fits')\n",
+    "    \n",
+    "print(namelist)\n",
+    "print(outlist)\n",
+    "\n",
+    "files  = namelist \n",
+    "outfile = outlist \n",
+    "\n",
+    "\n",
+    "for k, filename in enumerate(files):\n",
+    "    Detector1Pipeline.call(filename, config_file=calwebb_detector1_file, \n",
+    "                           output_file=outfile[k]);\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Here we run the second stage of the pipeline."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# gather the F150W yaml files back in the working directory\n",
+    "cwd = os.getcwd()\n",
+    "filter_pattern = os.path.join(cwd, 'yaml', filtname.lower(), 'jw*yaml') \n",
+    "files = glob(filter_pattern)[:] # Use [0:2] to test the first two files.\n",
+    "inputyaml = files\n",
+    "namelist = []\n",
+    "outlist = []"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Executing calwebb_image2"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "We select the *_rate.fits files and execute calwebb_image2 to generate the *_cal.fits files."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "for input_yaml_path in inputyaml:\n",
+    "    print(input_yaml_path)\n",
+    "    base = os.path.basename(input_yaml_path)\n",
+    "    name = os.path.splitext(base)[0]\n",
+    "   \n",
+    "    exi = os.path.exists(name + '_cal.fits')\n",
+    "    if exi:\n",
+    "        continue\n",
+    "\n",
+    "    namelist.append(name + '_rate.fits')\n",
+    "    outlist.append(name + '.fits')\n",
+    "        \n",
+    "print(namelist)\n",
+    "    \n",
+    "files = namelist \n",
+    "outfile = outlist "
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Generate the _cal.fits files."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "for k, filename in enumerate(files):\n",
+    "    print('-'*60)\n",
+    "    print(filename)\n",
+    "\n",
+    "    #here we assign the distortion file according to the detector\n",
```

Comment:
I don't think overriding the distortion reference file is necessary here. A couple years ago when there was a problem with the reference file, it was necessary. But the reference files currently in CRDS are correct. I think it would make more sense to allow the pipeline to query CRDS and automatically select the appropriate distortion reference file. This way, the old distortion reference files will not have to be stored and then downloaded by this notebook's users. Also, the distortion reference files will be updated during commissioning based on on-sky observations. At that point, the reference files currently being used here will be outdated and incorrect.

Note that the distortion reference file used to create the data in Mirage must be the same as the distortion reference file used by the pipeline during calibration. So the notebook that is used to call Mirage and create the input data for this notebook must be updated as well. If both notebooks are querying CRDS to find the appropriate reference files, then they will always be consistent with one another without having to worry about keeping a separate set of up-to-date reference files for downloading.